### PR TITLE
Elasticsearch: reduce code duplication

### PR DESCRIPTION
**Pull request by mantas-sidlauskas**
**What changed?**
* Removing code duplication form ES clients. 
* Bulk service related code is moved to a separate file. 
* TLS and AWS signing clients are initialized in a single place

**Why?**
To introduce official Elasticsearch client or OpenSearch client, we need to hide client details.
This PR is first of the refactoring series. For easier code review, I will make multiple and small changes. 

**How did you test it?**
Unit tests

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: common/elasticsearch/interfaces.go [hunk 1]
```diff
@@ -40,11 +41,30 @@ func NewGenericClient(
...
+		tlsClient, err = buildTLSHTTPClient(connectConfig.TLS)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	switch connectConfig.Version {
 	case "v6":
-		return NewV6Client(connectConfig, logger)
+		return NewV6Client(connectConfig, tlsClient, signingAWSClient, logger)
```
Comment (Shaddoll) [inline common/elasticsearch/interfaces.go:65]: the order is wrong

File: common/elasticsearch/common.go [hunk 2]
```diff
@@ -67,3 +84,23 @@ func GetESDocDelimiter() string {
...
+}
+
+func signingClientFromEnv(credentialConfig config.AWSEnvironmentCredential) (*http.Client, error) {
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(credentialConfig.Region)},
+	)
+	if err != nil {
+		return nil, err
+	}
+	return esaws.NewV4SigningClient(sess.Config.Credentials, credentialConfig.Region), nil
```
Comment (Shaddoll) [inline common/elasticsearch/common.go:105]: v6 and v7 are using different esaws packages

File: common/elasticsearch/client_v7.go [hunk 3]
```diff
@@ -221,73 +166,46 @@ func (c *elasticV7) ScanByQuery(ctx context.Context, request *ScanByQueryRequest
...
 
 	isLastPage := false
 	if err == io.EOF { // no more result
 		isLastPage = true
-		if scrollService != nil {
-			err := scrollService.Clear(ctx)
-			if err != nil {
-				c.logger.Warn("scrollService Clear fail", tag.Error(err))
-			}
+		if err := c.clearScroll(ctx, token.ScrollID); err != nil {
```
Comment (Shaddoll) [inline common/elasticsearch/client_v7.go:174]: the input should be `searchResult.ScrollID`

File: common/elasticsearch/interfaces.go (hunk 1 above)
Comment (mantas-sidlauskas) [reply common/elasticsearch/interfaces.go:65]: thanks!

File: common/elasticsearch/client_v7.go (hunk 3 above)
Comment (mantas-sidlauskas) [reply common/elasticsearch/client_v7.go]: good catch, thanks

File: common/elasticsearch/common.go (hunk 2 above)
Comment (mantas-sidlauskas) [reply common/elasticsearch/common.go:105]: they do, but all the code is the same. 

Will provide a local signer to remove dependency on olivere es client

**Review by Shaddoll — APPROVED** [review]

**Pull request merged**